var/spack/repos/builtin/packages/lbann/package.py

```diff
@@ -357,181 +326,179 @@ def setup_build_environment(self, env):
             env.append_flags("CXXFLAGS", self.spec["llvm-openmp"].headers.include_flags)
             env.append_flags("LDFLAGS", self.spec["llvm-openmp"].libs.ld_flags)
 
-    # Get any recent versions or non-numeric version
-    # Note that develop > numeric and non-develop < numeric
+    def _get_sys_type(self, spec):
+        sys_type = spec.architecture
+        if "SYS_TYPE" in env:
+            sys_type = env["SYS_TYPE"]
+        return sys_type
```

Comment:
I see several `CachedCMakePackage`s use this: I assume they could be simplified accordingly.

Reply:
(to clarify: I'm not requesting that happens in this PR)

Reply:
@scheibelp I think that this is a convention, so I am not sure that it should be moved up to the `cached_cmake.py` package.